### PR TITLE
Add documentation in the readme.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,122 @@
...
+This library exposes a unified, portable API for run-time reflection in Scala.
+It currently supports Scala/JVM and Scala.js, but will eventually support Scala Native as well (currently blocked on [scala-native#1279](https://github.com/scala-native/scala-native/issues/1279)).
+To be portable, only the subset of reflection capabilities that is implementable across all platforms is exposed.
+
+## Setup
+
+Add the following line to your (cross-)project's settings in `build.sbt`:
+
+```scala
+libraryDependencies += "org.portable-scala" %%% "reflect" % "0.1.0-SNAPSHOT"
```
Comment (gzm0) [inline README.md]: don't forget to update the artifact name

File: README.md [hunk 2]
```diff
@@ -1,3 +1,122 @@
...
+Once you have an `InstantiatableClass`, you can use its methods to instantiate the class:
+
+* `InstantiatableClass`
+  * `val runtimeClass: Class[_]`: the `java.lang.Class[_]` representing the class
+  * `val declaredConstructors: List[InvokableConstructor]`: a list of all invokable constructors of the class
+  * `def newInstance(): Any`: invokes the no-argument constructor of the class to create an instance
+  * `def getConstructor(parameterTypes: Class[_]*): Option[InvokableConstructor]`: find an invokable constructor by the types of its paramters
+* `InvokableConstructor`
+  * `val parameterTypes: List[Class[_]]`: the types of the parameters of that constructor
+  * `newInstance(args: Any*): Any`: invokes the constructor with the given actual arguments (which must conform to the parameter types)
```
Comment (gzm0) [inline README.md]: I would replace all this with an example. The Scaladoc is better suited for listing existing methods (and will not get out of date).

File: README.md [hunk 3]
```diff
@@ -1,3 +1,122 @@
...
+val cls = Reflect.lookupLoadableModuleClass("fully.qualified.ObjectName$", someClassLoader)
+```
+
+The `$` at the end of the object name is required.
+
+Once you have a `LoadableModuleClass`, you can use its methods to load the singleton instance of the object:
+
+* `LoadableModuleClass`
+  * `val runtimeClass: Class[_]`: the `java.lang.Class[_]` representing the module class
+  * `def loadModule(): Any`: loads the singleton instance of the object
```
Comment (gzm0) [inline README.md]: Same here. Just make an example.

**Review by gzm0 — CHANGES_REQUESTED** [review]

**Comment by sjrd** [comment]
Updated.

**Review by gzm0 — APPROVED** [review]

**Pull request merged**